Assets/MixedRealityToolkit/Providers/BaseInputDeviceManager.cs

```diff
@@ -61,36 +183,168 @@ public abstract class BaseInputDeviceManager : BaseDataProvider<IMixedRealityInp
         /// <param name="useSpecificType">Only register pointers with a specific type.</param>
         protected virtual IMixedRealityPointer[] RequestPointers(SupportedControllerType controllerType, Handedness controllingHand)
         {
-            var pointers = new List<IMixedRealityPointer>();
-
-            if ((Service != null) &&
-                (InputSystemProfile != null) &&
-                InputSystemProfile.PointerProfile != null)
+            using (RequestPointersPerfMarker.Auto())
             {
-                for (int i = 0; i < InputSystemProfile.PointerProfile.PointerOptions.Length; i++)
+                var returnPointers = new List<IMixedRealityPointer>();
+
+                CleanActivePointers();
+
+                for (int i = 0; i < pointerConfigurations.Length; i++)
                 {
-                    var pointerProfile = InputSystemProfile.PointerProfile.PointerOptions[i];
+                    var option = pointerConfigurations[i].profile;
+                    if (option.ControllerType.HasFlag(controllerType) && option.Handedness.HasFlag(controllingHand))
+                    {
+                        IMixedRealityPointer requestedPointer = null;
+
+                        if (EnablePointerCache)
+                        {
+                            var pointerCache = pointerConfigurations[i].cache;
+                            while (pointerCache.Count > 0)
+                            {
+                                var p = pointerCache.Pop();
```

Comment:
Why would we ever have more than one pointer of a given controller, handedness in the stack here?

Reply:
Handedness can equal "any" or both etc. The cache works per pointer option which is mostly 1:1 with a pointer prefab (not a hard expectation). 

![image](https://user-images.githubusercontent.com/25975362/71424813-403b7180-2653-11ea-8bfc-26d01fdf4c26.png)